nucleof072rb/Core/Src/main.c

```diff
@@ -87,14 +89,30 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  // bring CS pin to high
+  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	uint16_t read = handle_adc();
+
+	// 5% <= duty cycle <= 10%
+	// cycle*0.05 + (cycle*0.1-cycle*0.05)*(adc_val/(max val of 10 bits))
+	uint16_t pwm_val = (htim1.Init.Period*0.05)+ ((htim1.Init.Period*0.05)*(read/1023));
```

Comment:
`(read/1023)` will result in an undesired number. The hint is that they are both integer type number.